### PR TITLE
Make class attributes to be a list

### PR DESCRIPTION
**Pull request by agrawalsourav98**
Made class attributes parameter in `convert_geojson.py` to be a list to allow use of geojsons which have different feature columns representing class.

Use Case, The latest labelling has`class`, `Class`  and `id` all representing the class of a defect.

### CODE REVIEW

File: ml_dronebase_data_utils/geo/convert_geojson.py [hunk 1]
```diff
@@ -53,7 +53,15 @@ def geo_to_voc(
             boxes = vertices_to_boxes(vertices)
 
         if class_attribute is not None:
-            names = gdf[class_attribute]
+            for attr in class_attribute:
+                try:
+                    names = gdf[attr]
+                except KeyError:
+                    pass
+            if not len(names):
```
Comment (isaaccorley) [inline ml_dronebase_data_utils/geo/convert_geojson.py:61]: Is this indented properly?

File: ml_dronebase_data_utils/geo/convert_geojson.py [hunk 2]
```diff
@@ -53,7 +53,15 @@ def geo_to_voc(
             boxes = vertices_to_boxes(vertices)
 
         if class_attribute is not None:
-            names = gdf[class_attribute]
+            for attr in class_attribute:
```
Comment (isaaccorley) [inline ml_dronebase_data_utils/geo/convert_geojson.py:56]: This for loop can be replaced with something like:

if not all([True if gdf.get(attr) else False for attr in class_attributes]):
   raise ValueError(...)

Try/except is not a good solution in this case.

**Review by isaaccorley — CHANGES_REQUESTED** [review]

File: ml_dronebase_data_utils/geo/convert_geojson.py (hunk 2 above)
Comment (agrawalsourav98) [reply ml_dronebase_data_utils/geo/convert_geojson.py:56]: Okay, the main idea is to find the attribute from the list which applies to the current gdf objejct. I will replace try/catch with something like a for loop which checks which one matches and use that instead.

File: ml_dronebase_data_utils/geo/convert_geojson.py [hunk 3]
```diff
@@ -53,7 +53,14 @@ def geo_to_voc(
...
+                names = gdf.get(attr, [])
```
Comment (isaaccorley) [inline ml_dronebase_data_utils/geo/convert_geojson.py:57]: Can you explain what the goal of this for loop is? It's not clear to me why you are defaulting gdf.get to return an empty list when it naturally defaults to None. Are you just checking to make sure the class attributes are valid classes that user defines in the CLI call?

**Review by isaaccorley — CHANGES_REQUESTED** [review]

File: ml_dronebase_data_utils/geo/convert_geojson.py (hunk 3 above)
Comment (agrawalsourav98) [reply ml_dronebase_data_utils/geo/convert_geojson.py:57]: So, the for loop is for handling multiple class attributes. Say, you have 100 geojsons and they have the defect class represented by attributes `A`, `B` and `C`, so the for loop would check which attribute is present in the particular geojson and uses that to get the classes from a particular geojson.
If either of the three is not found if errors out, so the user can ideally add it and then re-run the tool.

Comment (agrawalsourav98) [reply ml_dronebase_data_utils/geo/convert_geojson.py:57]: For the empty list thing, I have only added it to represent the list of classes were not found using the given attribute.

**Pull request merged**